domain/src/test/java/org/oppia/android/domain/oppialogger/analytics/FeatureFlagsLoggerTest.kt

```diff
@@ -0,0 +1,183 @@
+import org.oppia.android.testing.logging.EventLogSubject.Companion.assertThat
+import org.oppia.android.testing.logging.SyncStatusTestModule
+import org.oppia.android.testing.platformparameter.TestPlatformParameterModule
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestCoroutineDispatchers
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.caching.AssetModule
+import org.oppia.android.util.data.DataProvidersInjector
+import org.oppia.android.util.data.DataProvidersInjectorProvider
+import org.oppia.android.util.locale.LocaleProdModule
+import org.oppia.android.util.logging.EnableConsoleLog
+import org.oppia.android.util.logging.EnableFileLog
+import org.oppia.android.util.logging.GlobalLogLevel
+import org.oppia.android.util.logging.LogLevel
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.platformparameter.TEST_FEATURE_FLAG
+import org.oppia.android.util.platformparameter.TEST_FEATURE_FLAG_WITH_ENABLED_DEFAULTS
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [FeatureFlagsLogger]. */
+// FunctionName: test names are conventionally named with underscores.
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = FeatureFlagsLoggerTest.TestApplication::class)
+class FeatureFlagsLoggerTest {
+  private companion object {
+    private const val TEST_SESSION_ID = "test_session_id"
+  }
+
+  @Inject lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+  @Inject lateinit var featureFlagsLogger: FeatureFlagsLogger
+  @Inject lateinit var fakeAnalyticsEventLogger: FakeAnalyticsEventLogger
+
+  @Before
+  fun setup() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testLogFeatureFlags_logFeatureFlags_hasCorrectSessionId() {
+    featureFlagsLogger.logAllFeatureFlags(TEST_SESSION_ID)
+    testCoroutineDispatchers.runCurrent()
+
+    val eventLog = fakeAnalyticsEventLogger.getMostRecentEvent()
+
+    assertThat(eventLog).hasFeatureFlagContextThat {
+      hasSessionIdThat().isEqualTo(TEST_SESSION_ID)
+    }
+  }
+
+  @Test
+  fun testLogFeatureFlags_logsTestFeatureFlag_hasCorrectDefaultValues() {
+    featureFlagsLogger.logAllFeatureFlags(TEST_SESSION_ID)
+    testCoroutineDispatchers.runCurrent()
+
+    val eventLog = fakeAnalyticsEventLogger.getMostRecentEvent()
+
+    assertThat(eventLog).hasFeatureFlagContextThat {
+      val testFeatureFlag = hasFeatureFlagThat(TEST_FEATURE_FLAG)
+
+      assertFeatureFlag(testFeatureFlag).isNotNull()
+      featureFlagHasEnabledStateThat(testFeatureFlag).isEqualTo(false)
+      featureFlagHasSyncStatusThat(testFeatureFlag).isEqualTo(SyncStatus.NOT_SYNCED_FROM_SERVER)
+    }
+  }
```

Comment:
```suggestion
  fun testLogFeatureFlags_logsTestFeatureFlag_hasCorrectDefaultValues() {
    featureFlagsLogger.logAllFeatureFlags(TEST_SESSION_ID)
    testCoroutineDispatchers.runCurrent()

    val eventLog = fakeAnalyticsEventLogger.getMostRecentEvent()

    assertThat(eventLog).hasFeatureFlagContextThat {
     hasFeatureFlagThatListThat().hasFeatureFlagItemContextThat(){
    hasNameThat().isEqualTo("Some name")
    hasEnabledStateThat().isEqualTo(false)
    }
hasSyncStatusThat().isEqualTo(SyncStatus.NOT_SYNCED_FROM_SERVER)
    }
  }
```

You will need to create a `FeatureFlagItemContext` that has `hasFeatureFlagItemContextThat()` to use in your assertions.
So the assertions will be nested.

Reply:
I modified the tests to reflect the new structure.